### DISCUSSION TITLE
parsing of 'patrol test' arguments

### DISCUSSION THREAD

**Discussion by proximilian** [leancodepl/patrol#1009] [Q&A]
Hello Guys,
I have several flavors of an app. The different flavors use different package_names and are build by different main.dart files. When I start my tests with `patrol test --target .... --flavor sit` I want my tests automatically to get that flavor argument and to use it to define the correct package_names and to build the correct main.dart file according to the declared flavor in the command line. 
Does somebody have an idea how to get that argument in an easy way? I tried with the ArgParser package (https://pub.dev/packages/args/install)  but unsuccessfully or possibly I am making a mistake somewhere. Thanks in advance for your time.

**Comment by bartekpacia** [comment]
Interesting use case. We don't currently support it, so all you can do is write some custom shell script around `patrol_cli` to accomplish what you want.

Do you have a different flavor for _every_ `*_test.dart` file (i.e is it a 1:1 relation)?

**Reply by proximilian** [reply]
Thanks for your quick answer, Bartek.
 Yes, it can be different for every test, but I don't plan to test it like that. I plan to run all (or part of) test once for the one flavor and then run the same tests for the another one. But as I said above I want to implement a method that defines the particular package_name and main.dart file relevant to the flavor argument declared in the command line. Do you have some ideas for the script?

**Reply by bartekpacia** [reply]
I think you have to write such a small command-line tool yourself :)

**Comment by proximilian** [comment]
Two more question @bartekpacia in this line of thinking: 
- Can I list several flavors and specify for each of them a different package_name (bundle_id) in the Patrol section at pubspec.yaml? Or I can only specify one flavor there?
- I wonder if I have declared the package_name, androidAppName etc. at the NativeAutomatorConfig, is it mandatory to have the Patrol section with the same information at the pubspec.yaml?

**Reply by bartekpacia** [reply]
> Can I list several flavors and specify for each of them a different package_name (bundle_id) in the Patrol section at pubspec.yaml? Or I can only specify one flavor there?

Nope. Best you can do is specify a different flavor for Android and iOS, e.g:


```yaml
patrol:
  app_name: Awesome App
  android:
    package_name: pl.leancode.awesomeapp
    flavor: development
  ios:
    bundle_id: pl.leancode.AwesomeApp
    app_name: The Awesome App
    flavor: devIOS
```

> I wonder if I have declared the package_name, androidAppName etc. at the NativeAutomatorConfig, is it mandatory to have the Patrol section with the same information at the pubspec.yaml?

You don't have to do anything with `NativeAutomatorConfig`. These options are there in case you don't run tests with `patrol_cli`, but the recommended way is to use `patrol_cli` and specify these values in `pubspec.yaml` only.

**Reply by proximilian** [reply]
Тhanks for the detailed answer. 💯 
If you find my first question (allowing declaration of several flavors with different packages in pubspec.yaml) reasonable and worth implementing in future, I will highly appreciate it.

**Reply by bartekpacia** [reply]
Happy to help :)

I think it's interesting. Would like to create a new feature-request issue and describe it in more detail? Would be awesome if you included more examples of what you want 🙏

You can also add a link to this discussion :)

**Comment by proximilian** [comment]
A new feature-request is created as a result of the current discussion --> #1049
